src/python/py_generate_pipeline.cpp

```diff
@@ -534,4 +549,61 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
         .def(py::init<>())
         .def("put", &StreamerBase::put)
         .def("end", &StreamerBase::end);
+
+    py::class_<GenerationResult>(m, "GenerationResult")
+        .def(py::init<>())
+        .def_readonly("m_request_id", &GenerationResult::m_request_id)
+        .def_property("m_generation_ids",
+            [](GenerationResult &r) -> py::list {
+                py::list res;
+                for (auto s: r.m_generation_ids) {
+                    PyObject* py_s = PyUnicode_DecodeUTF8(s.data(), s.length(), "replace");
```

Comment:
Can this fail? Shall we do something specific if it does?

Reply:
I believe no. `replace` indicates that if unicode point fails to be decoded it must be replaced with ?. Anyway, I just copied it from python.cpp